OsmAnd/src/net/osmand/plus/plugins/mapillary/MapillaryVectorLayer.java

```diff
@@ -456,4 +553,16 @@ public int getRadius(RotatedTileBox tb) {
 		}
 		return (int) (r * view.getScaleCoefficient());
 	}
+
+	private long getFilterKey() {
+		int hasFilter = plugin.USE_MAPILLARY_FILTER.get() ? 1 : 0;
+		long from = 0;
+		long to = 0;
+		if (hasFilter == 1) {
+			from = plugin.MAPILLARY_FILTER_FROM_DATE.get();
+			to = plugin.MAPILLARY_FILTER_TO_DATE.get();
+		}
+		int pano = plugin.MAPILLARY_FILTER_PANO.get() ? 1 : 0;
+		return (hasFilter << 1) + from + to + pano;
```

Comment:
Bad key. If user will decrease **from** by 1 and increase **to** by 1 - nothing changes.

Reply:
From and to in millisec. No bug